ci/PIPELINE.md

```diff
@@ -0,0 +1,117 @@
+
+or with tekton cli
+```sh
+tkn pipeline start kogito-serverless-operator-pipeline \
+-w name=shared-workspace,volumeClaimTemplateFile=https://raw.githubusercontent.com/kiegroup/kogito-serverless-operator/main/ci/volume/persistent_volume.yaml \
+-p deployment-name=kogito-serverless-operator \
+-p git-url=https://github.com/kiegroup/kogito-serverless-operator.git \
+-p git-revision=main \
+-p IMAGE='image-registry.openshift-image-registry.svc:5000/kogito-serverless-operator-system/kogito-serverless-operator:latest' \
+--use-param-defaults
+```
+
+8) Checks Pipeline execution
+ 
+Open the Pipeline Menu under the namespace/project kogito-serverless-operator-system
+or with tekton cli with the id printed starting the pipelinerun from the cli
+
+```sh
+tkn pipelinerun logs kogito-serverless-operator-pipeline-run-<id> -f -n <your-namespace>
+```
+
+### How to see the content of the workspace
+1) Create the task show_workspace_content
+```sh
+kubectl apply -f ci/task/show_workspace_content.yaml
+```
+
+2) Add the task show-workspace in the pipeline after the fetch-repository or build-image
+
+### How to deploy again 
+Go to the pipeline runs and ask on a already executed pipelinerun a rerun
+
+
+## Trigger the pipeline on GithubEvents
+
+1) Create the trigger binding
+```sh
+oc create -f ci/pipeline/trigger/trigger_binding.yaml
+```
+2) Create the trigger template
+```sh
+oc create -f ci/pipeline/trigger/trigger_template.yaml
+```
+
+3) Create the trigger resource
+```sh
+oc create -f ci/pipeline/trigger/trigger_resource.yaml
+```
+
+4) Create the Event listener trigger
+```sh
+oc create -f ci/pipeline/trigger/trigger_event_listener.yaml
+```
+
+5) Check the webhook
+```sh
+tkn el -n kogito-serverless-operator-pipeline ls
+```
+
+```sh
+NAME                                 AGE              URL                                                                                                       AVAILABLE
+kogito-serverless-operator-webhook   16 minutes ago   http://el-kogito-serverless-operator-webhook.kogito-serverless-operator-pipeline.svc.cluster.local:8080   True
+```
+
+```sh
+kubectl get pods,svc -n kogito-serverless-operator-pipeline -l eventlistener=kogito-serverless-operator-webhook
+```
+```sh
+NAME                                                         READY   STATUS    RESTARTS   AGE
+pod/el-kogito-serverless-operator-webhook-7f64bc86db-kckqb   1/1     Running   0          17m
+
+NAME                                            TYPE        CLUSTER-IP     EXTERNAL-IP   PORT(S)             AGE
+service/el-kogito-serverless-operator-webhook   ClusterIP   10.217.5.248   <none>        8080/TCP,9000/TCP   17m
+```
+
+7) Create Event Listener Ingress
+```sh
+   oc expose svc -n kogito-serverless-operator-pipeline el-kogito-swf-operator-webhook
+   export EL_WEBHOOK_HOST=$(oc get route el-gitea-webhook -n triggers-demo -ojsonpath='{.spec.host}')
+```
+8) Add a webhook in your github/gitlab repo with the url of the listener on openshift
```

Comment:
no webhook secret ?
I would suggest it for security reasons

Reply:
@radtriste  Do we want to enable the triggers in a internal cluster ?

Reply:
Ok, I'll add this section of the doc https://docs.openshift.com/container-platform/4.12/cicd/pipelines/securing-webhooks-with-event-listeners.html